inferno-ml-server/src/Inferno/ML/Server.hs

```diff
@@ -72,19 +72,19 @@ main = runServer =<< mkOptions
         mkSettings :: (Request -> Status -> Maybe Integer -> IO ()) -> Settings
         mkSettings logger =
           defaultSettings
-            & setPort (view #port cfg & fromIntegral)
+            & setPort (fromIntegral cfg.port)
             & setLogger logger
 
 runInEnv :: Config -> (Env -> IO ()) -> IO ()
 runInEnv cfg f = withRemoteTracer $ \tracer -> do
   traceWith tracer $ InfoTrace StartingServer
-  withConnect (view #store cfg) $ \conn ->
-    f
-      =<< Env cfg conn tracer
-        <$> newMVar ()
-        <*> newEmptyMVar
-        <*> newManager defaultManagerSettings
-        <*> newIORef Nothing
+  f
+    =<< Env cfg tracer
+      <$> newConnectionPool cfg.store
```

Comment:
It might be reasonable to destroy the pool when done so it closes connections explicitly. With sqlite, failing to do so left dangling WAL files which are annoying. With postgres it may leave open resources in the db server until a timeout expires (not sure but it wouldn’t surprise me)

Reply:
Good idea, I've added that